### PR TITLE
Bindplane agent terraform

### PR DESCRIPTION
**Pull request by jsirianni**
Please not that this PR also includes the changes from this open PR https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/pull/33

- Added terraform/agent/bindplane directory, which will hold all Terraform examples for the BindPlane agent
- Added Terraform + Linux BindPlane agent example
- Added Terraform + Windows BindPlane agent example

### CODE REVIEW

**Comment by xiangshen-dk** [comment]
Thanks for your PRs! I will merge them after review.

File: terraform/agents/bindplane/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+This directory serves as an example on how to leverage Terraform's [template_file](https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file) to deploy the BindPlanbe Agent to Linux and Windows.
+
+[BindPlane](https://docs.bindplane.bluemedora.com/docs) is a service that Google Cloud user's can use to stream metrics and logs to Google Cloud. BindPlane does not add any additional cost, you are billed by Google for your metric / log ingestion the same way you would be if you used the Cloud Ops agent. [Click here to learn more](https://cloud.google.com/blog/products/management-tools/use-bluemedoras-bindplane-with-google-cloud).
+
+## Prerequisite
+
+- [BindPlane Account](https://bindplane.bluemedora.com/)
+- BindPlane secret key
+  - Go to Univeral Agents, Add agent, Linux, Copy the "SECRET_KEY" value
+- [Gcloud SDK](https://cloud.google.com/sdk/docs/install)
```
Comment (xiangshen-dk) [inline terraform/agents/bindplane/README.md:12]: Google Cloud SDK

File: terraform/agents/bindplane/scripts/install.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# exit early if bpagent is already installed, otherwise it will install
+# over itself and polute the agents page in the BindPlane UI
+sudo systemctl status bpagent
+if [ $? = 0 ]; then
+    # mysqld service exists
```
Comment (xiangshen-dk) [inline terraform/agents/bindplane/scripts/install.sh:20]: mysqld service?

File: terraform/agents/bindplane/windows.tf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+Caution: Windows requires more resources than Linux, this example uses a larger
+instance type and ssd storage.
+
+**/
+
+resource "google_compute_instance" "windows2019" {
+  project      = var.project
+  name         = "windows-2019-bindplane-agent"
+  machine_type = "n1-standard-1"
```
Comment (xiangshen-dk) [inline terraform/agents/bindplane/windows.tf:35]: Maybe you can change N1 to N2.

Comment (jsirianni) [reply terraform/agents/bindplane/windows.tf]: Resolved

File: terraform/agents/bindplane/scripts/install.sh (hunk 2 above)
Comment (jsirianni) [reply terraform/agents/bindplane/scripts/install.sh]: Good catch, thank you.

File: terraform/agents/bindplane/README.md (hunk 1 above)
Comment (jsirianni) [reply terraform/agents/bindplane/README.md]: resolved

**Pull request merged**